### PR TITLE
Sample app with Sovereign Cloud support

### PR DESCRIPTION
**Pull request by iambmelt**
This PR adds support to the sample app for sovereign cloud

Authority: `AAD_COMMON`
Login hint: **Must be blank** (AAD issue)
Resource: `SIMPLE`
Client id: `BLACKFOREST`
Extra qp: `instance_aware=true`

### CODE REVIEW

**Comment by heidijinxujia** [comment]
LGTM so far, I will try with it!

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 1]
```diff
@@ -60,7 +60,8 @@ public String getText() {
     enum ClientId {
         MSDEVEX("b92e0ba5-f86e-4411-8e18-6b5f928d968a"),
         ONEDRIVE("af124e86-4e96-495a-b70a-90f90ab96707"),
-        OFFICE("d3590ed6-52b3-4102-aeff-aad2292ab01c");
+        OFFICE("d3590ed6-52b3-4102-aeff-aad2292ab01c"),
+        BLACKFOREST("f5d01c1c-abe6-4207-ae2d-5bc9af251724");
```
Comment (addev-ashish) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java]: Can you rename to AppCheck2_BF or something like that, we might have more ClientId from them in the future

Comment (iambmelt) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java]: Done 👍

**Comment by heidijinxujia** [comment]
@iambmelt I've merged my current change on authority updating of sample app into the PR.

**Comment by iambmelt** [comment]
@heidijinxujia Oooph. Yes, let's call this a bug. IMHO the sample should demonstrate usage of the app in all scenarios. Is this sample app _just_ for test, or do we plan to make this the official new sample?

If this should get fixed, doesn't this affect usage in any scenario where interactive auth happens and there's a possibility that the user account changes between sov/non-sov?

As a side question, it never was clear to me why auto-updating the authority was out-of-scope for the sov-cloud changes? Does anybody remember?

/cc @addev-ashish

**Comment by iambmelt** [comment]
Per team mtg - sov cloud / non-sov cloud account switching is an unsupported scenario atm and changing sample is out of scope.

I'll open an issue (P2) and link back here.

/cc @heidijinxujia @addev-ashish

**Pull request merged**